### PR TITLE
[CBRD-23287] Fix notifying task capper of ended task

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-23287

Fix task being finished without notifying capper.

Made it safer by automatically wrapping normal task with a worker_pool_task_capper::capped_task that calls end_task at the end of execution.

Removed manual end task call.

### CODE REVIEW

File: src/loaddb/load_session.cpp [hunk 1]
```diff
@@ -121,7 +121,7 @@ namespace cubload
 	thread_ref.conn_entry = &m_conn_entry;
 	driver *driver = thread_ref.m_loaddb_driver;
 
-	assert (driver != NULL && !driver->is_initialized ());
+	assert (driver != NULL &&!driver->is_initialized ());
```
Comment (valerinistor) [inline src/loaddb/load_session.cpp]: revert this line to original state

File: src/thread/thread_worker_pool_taskcap.hpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  template <typename Context>
+  void
+  worker_pool_task_capper<Context>::capped_task::execute (context_type &ctx)
+  {
+    m_nested_task->execute (ctx);
+    m_capper.end_task ();
+  }
+}
+
+#endif _THREAD_WORKER_POOL_TASKCAP_HPP_
```
Comment (valerinistor) [inline src/thread/thread_worker_pool_taskcap.hpp]:
```suggestion
#endif /* _THREAD_WORKER_POOL_TASKCAP_HPP_ */
```

File: src/thread/thread_worker_pool_taskcap.hpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  //////////////////////////////////////////////////////////////////////////
+  template <typename Context>
+  worker_pool_task_capper<Context>::worker_pool_task_capper (worker_pool<Context> *wp)
+  {
+    m_worker_pool = wp;
+    m_tasks_available = m_max_tasks = wp->get_max_count ();
+  }
+
+  template <typename Context>
+  worker_pool_task_capper<Context>::~worker_pool_task_capper ()
```
Comment (valerinistor) [inline src/thread/thread_worker_pool_taskcap.hpp]: use '= default' to define a trivial destructor

File: src/loaddb/load_session.cpp (hunk 1 above)
Comment (eseokoh) [inline src/loaddb/load_session.cpp]: please bring the space back.

File: src/thread/thread_worker_pool_taskcap.hpp (hunk 2 above)
Comment (eseokoh) [inline src/thread/thread_worker_pool_taskcap.hpp]:
```suggestion
#endif // _THREAD_WORKER_POOL_TASKCAP_HPP_
```

File: src/thread/thread_worker_pool_taskcap.hpp [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  };
+} // namespace cubthread
+
+namespace cubthread
+{
+  //////////////////////////////////////////////////////////////////////////
+  // worker_pool_task_capper template implementation
+  //////////////////////////////////////////////////////////////////////////
+  template <typename Context>
+  worker_pool_task_capper<Context>::worker_pool_task_capper (worker_pool<Context> *wp)
```
Comment (valerinistor) [inline src/thread/thread_worker_pool_taskcap.hpp]: please use same parameter name as in c-tor declaration

File: src/thread/thread_worker_pool_taskcap.hpp [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      using task_type = task<Context>;
+      using worker_pool_type = worker_pool<Context>;
+
+    public:
+      worker_pool_task_capper (worker_pool<Context> *worker_pool);
+      ~worker_pool_task_capper ();
+
+      void push_task (task<Context> *task);
+      cubthread::worker_pool<Context> *get_worker_pool ();
+      void end_task ();
```
Comment (valerinistor) [inline src/thread/thread_worker_pool_taskcap.hpp]: I think better will be to limit end_task function to private access, in order to avoid being called

**Pull request merged**